Packs/SentinelOne/Integrations/SentinelOne-V2/SentinelOne-V2.py

```diff
@@ -3181,57 +3235,116 @@ def get_mirroring_fields(params):
+    incidents_threats = []
+    current_fetch = args.get('current_fetch')
 
-    threats = client.get_threats_request(limit=fetch_limit, created_after=last_fetch_date_string, site_ids=fetch_site_ids)
+    threats = client.get_threats_request(limit=args.get('fetch_limit'),
+                                         created_after=args.get('last_fetch_date_string'),
+                                         site_ids=args.get('fetch_site_ids'))
     for threat in threats:
         rank = threat.get('rank')
-        threat.update(get_mirroring_fields(params))
         try:
             rank = int(rank)
         except TypeError:
             rank = 0
         # If no fetch threat rank is provided, bring everything, else only fetch above the threshold
-        if IS_VERSION_2_1 or rank >= fetch_threat_rank:
-            incident = threat_to_incident(threat)
+        if IS_VERSION_2_1 or rank >= args.get('fetch_threat_rank'):
+            threat['type'] = 'threat'
+            incident = to_incident('Threat', threat)
             date_occurred_dt = parse(incident['occurred'])
             incident_date = int(date_occurred_dt.timestamp() * 1000)
-            if incident_date > last_fetch:
-                incidents.append(incident)
+            if incident_date > int(args.get('last_fetch')):
+                incidents_threats.append(incident)
 
             if incident_date > current_fetch:
                 current_fetch = incident_date
 
+    return incidents_threats, current_fetch
+
+
+def fetch_alerts(client: Client, args):
+
+    incidents_alerts = []
+    current_fetch = args.get('current_fetch')
+
+    query_params = assign_params(
+        incidentStatus=','.join(args.get('fetch_incidentStatus')),
+        createdAt__gte=args.get('last_fetch_date_string'),
+        limit=args.get('fetch_limit'),
+        siteIds=args.get('fetch_site_ids')
+    )
+
+    alerts, pagination = client.get_alerts_request(query_params)
+    for alert in alerts:
+        severity = alert.get('ruleInfo').get('severity')
+
+        if str(severity) in args.get('fetch_severity'):
+            alert['type'] = 'alert'
+            incident = to_incident('Alert', alert)
+            date_occurred_dt = parse(incident['occurred'])
+            incident_date = int(date_occurred_dt.timestamp() * 1000)
+            if incident_date > args.get('last_fetch'):
+                incidents_alerts.append(incident)
+
+            if incident_date > current_fetch:
+                current_fetch = incident_date
+
+    return incidents_alerts, current_fetch
+
+
+def fetch_handler(client: Client, args):
+
+    last_run = demisto.getLastRun()
+    last_fetch = last_run.get('time')
+
+    if last_fetch is None:
+        last_fetch = dateparser.parse(args.get('first_fetch_time'), settings={'TIMEZONE': 'UTC'})
+        if not last_fetch:
+            raise DemistoException('Please provide an initial First fetch timestamp')
```

Comment:
instead if adding a condition to check if the fetch_time param was provided (which must be provided), make the "fetch_time" param required, its much better

Reply:
thought about this one. We will have customers that don't want to enable fetching on the instance. Therefor having the last_fetch a required field might be confusing. Would work if we can apply conditional parameters to make the field required only if fetching is enabled.